Sources/ApolloCodegenLib/Frontend/ApolloCodegenFrontend.swift

```diff
@@ -0,0 +1,94 @@
+import Foundation
+import JavaScriptCore
+
+public final class ApolloCodegenFrontend {
+  static let bundle = Bundle(for: ApolloCodegenFrontend.self)
+  private static let libraryURL = bundle.url(forResource: "ApolloCodegenFrontend.bundle", withExtension: "js")!
+  private static let librarySource = try! String.init(contentsOf: libraryURL)
+  
+  private let virtualMachine = JSVirtualMachine()
+  private let context: JSContext
+  private let bridge: JavaScriptBridge
+  
+  private let library: JavaScriptObject
+  
+  public init() throws {
+    let context = JSContext(virtualMachine: virtualMachine)!
+    self.context = context
+    bridge = JavaScriptBridge(context: context)
+    
+    try bridge.throwingJavaScriptErrorIfNeeded {
+      context.evaluateScript(Self.librarySource, withSourceURL: Self.libraryURL)
+    }
+    
+    library = bridge.fromJSValue(context.globalObject["ApolloCodegenFrontend"])
+    
+    bridge.register(GraphQLSource.self, forJavaScriptClass: "Source", from: library)
+    
+    bridge.register(GraphQLError.self, from: library)
+    bridge.register(GraphQLSchemaValidationError.self, from: library)
+    
+    bridge.register(GraphQLSchema.self, from: library)
+    
+    bridge.register(GraphQLScalarType.self, from: library)
+    bridge.register(GraphQLEnumType.self, from: library)
+    bridge.register(GraphQLInputObjectType.self, from: library)
+    bridge.register(GraphQLObjectType.self, from: library)
+    bridge.register(GraphQLInterfaceType.self, from: library)
+    bridge.register(GraphQLUnionType.self, from: library)
+  }
+  
+  public func loadSchema(from fileURL: URL) throws -> GraphQLSchema {
+    precondition(fileURL.isFileURL)
+    
+    if fileURL.pathExtension == "json" {
+      let introspectionResult = try String(contentsOf: fileURL)
+      return try loadSchemaFromIntrospectionResult(introspectionResult)
+    } else {
+      let source = try makeSource(from: fileURL)
+      return try loadSchemaFromSDL(source)
```

Comment:
I haven't seen `.sdl`, but `graphqls` is pretty common, so we should at least allow that.

Reply:
For what it's worth, Android uses `.sdl` by default to distinguish from operation files.

Reply:
If there's some consensus about `.graphqls`, we should probably switch to `.graphqls` by default. We can always keep `.sdl` compatibilty

Reply:
To be honest I've _only_ seen `.sdl` rather than `.graphqls` 🙃. I have no opposition to supporting both until someone declares a winner

Reply:
Yeah, I'm not sure if there's any consensus. I've never seen `.sdl`, but that's just based on my limited experience. We could ask the rest of the company so we can at least agree on a recommendation for our docs. But we should probably support both. Or always try to parse as SDL when it's not `.json`, because the worst that could happen is a parsing error.

Reply:
> Or always try to parse as SDL when it's not .json, because the worst that could happen is a parsing error.

That sounds reasonable at first glance, but I do think we want to at least provide some guidance in terms of what reasonably supported extensions are. I think supporting both `graphqls` and `sdl` is reasonable, though.

Reply:
For what it's worth, I blogged about `.sdl` there: https://www.apollographql.com/blog/sdl-support-in-apollo-android/

Happy to update it to use `.graphqls` if that makes more sense as it's definitely better to be consistent across our docs.

Reply:
I think for now let's leave this as is where it'll try to parse SDL for anything that isn't explicitly JSON.